dev-python/pyside-tools/pyside-tools-9999.ebuild

```diff
@@ -0,0 +1,118 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+#FIXME: Restore Python 2.7 support after this upstream issue is resolved:
+#    https://bugreports.qt.io/browse/PYSIDE-508
+#Even after this issue is resolved, further ebuild changes will probably be
+#needed to fully support Python 2.7. Unlike PySide2 and Shiboken2, the root
+#"CMakeLists.txt" file in pyside2-tools provides no ${PYTHON_EXECUTABLE} option.
+#Instead, the ${PYTHON_EXTENSION_SUFFIX} and ${PYTHON_BASENAME} options must be
+#passed with values specific to Python 2.7. Additionally, note that the
+#"python2.7-config" command provides no "--extension-suffix" option.
+PYTHON_COMPAT=( python3_{4,5,6} )
+# PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+
+inherit cmake-utils python-r1 virtualx git-r3
+
+DESCRIPTION="PySide development tools (lupdate, rcc, uic)"
+HOMEPAGE="https://wiki.qt.io/PySide2"
+EGIT_REPO_URI=(
+	"git://code.qt.io/pyside/${PN}.git"
+	"https://code.qt.io/git/pyside/${PN}.git"
+)
+#FIXME: Switch to the clang-enabled "dev" branch once stable.
+EGIT_BRANCH="5.6"
+
+# See the top-level "LICENSE-rcc" and "LICENSE-uic" files for licensing.
+LICENSE="BSD GPL-2"
+SLOT="2"
+KEYWORDS=""
+IUSE="test"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+# Minimum version of Qt required.
+QT_PV="5.6*:5"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-python/pyside:2[gui,${PYTHON_USEDEP}]
+	=dev-qt/qtcore-${QT_PV}
+	=dev-qt/qtgui-${QT_PV}
+"
+DEPEND="${RDEPEND}
+	test? ( virtual/pkgconfig )
+"
+
+src_prepare() {
+	cmake-utils_src_prepare
+
+	preparation() {
+		if python_is_python3; then
```

Comment:
**Oh!** So _that's_ why the official `pyside-tools` ebuild goes to such excruciatingly intense lengths to perform an in-source build. Since that ebuild is mostly undocumented and since the default out-of-source build behaved as expected, I assumed an in-source build was no longer necessary.

I assumed incorrectly. You're absolutely right: I only built `pyside-tools` against a single Python 3.x version and failed to notice **that it's completely broken for multiple Python 3.x versions.** `</sigh>`

Awesome catch. I'll have to contemplate how to correct this. I'd rather not revert to an in-source build if I can help it. Maybe I can't help it, of course. But maybe I can.

The test-specific `sed` commands can probably be safely segregated to the `src_test()` phase. Maybe? If so, that only leaves the removal of:

* The `pyside2uic/port_v2` subdirectory when building against Python 3.
* The `pyside2uic/port_v3` subdirectory when building against Python 2.

These removals can probably be safely segregated to the `src_install()` phase. Maybe? If so, the `src_prepare()` phase could then be entirely removed and the need for an in-source build goes away.

Maybe? Let's try.